c2corg_ui/views/document.py

```diff
@@ -413,20 +421,43 @@ def _get_api_cache_key_from_etag(self, etag):
     def _get_response(self, page_html):
         return get_response(self.request, page_html)
 
+    def _redirect(self, id, lang, slug=None):
+        location = ''
+        if slug is None:
+            location = self.request.route_url(
+                self._API_ROUTE + '_view_id_lang', id=id, lang=lang)
+        else:
+            location = self.request.route_url(
+                self._API_ROUTE + '_view', id=id, lang=lang, slug=slug)
+        raise HTTPMovedPermanently(location=location)
```

Comment:
> In order to actually transfer the indexing of the old page to the new one, a 301 is recommended.

If it really is a permanent redirect, yes. I am worried that accidentally a redirect is set to a wrong document. And then you have a hard undoing the (not so) permanent redirect.